confidant/routes/credentials.py

```diff
@@ -514,6 +522,7 @@ def revert_credential_to_revision(id, to_revision):
             cipher_version=revert_credential.cipher_version,
             modified_by=authnz.get_logged_in_user(),
             documentation=revert_credential.documentation,
+            last_rotation_date=revert_credential.last_rotation_date,
```

Comment:
For revert we also need to check whether or not the credentials differ. If someone reverts a credential, but it only changes the name or tags, for instance, the rotation time hasn't changed.

Reply:
But the rotation time of the credential should have already reflected that.  For instance:

```
version 1: (credential_pair: "asdf", last_rotation_date: T0, name: "the OG key")

# credential_pair changed so update last_rotation_date
version 2: (credential_pair: "bbbb", last_rotation_date: T1, name: "new key). 

# credential_pair is the same so don't update last_rotation_date
version 3: (credential_pair: "bbbb", last_rotation_date: T1, name: "still the new key")
```

Say version 3 is the current version and we want to revert back to version 2, `last_rotation_date: T1` is still accurate and *it's accurate because of our checks in the update credential endpoint*.  

The update endpoint essentially creates an invariant where `last_rotation_date` reflects the time the key was actually changed, not when the object was updated.  This endpoint just takes advantage of that invariant.

Reply:
Ah. Good point.